src/lib/zkapp.ts

```diff
@@ -1036,21 +1036,36 @@ super.init();
   }
 
   /**
-   * Fetches a list of events that have been emitted by this {@link SmartContract}.
-   */
+    Asynchronously fetches events emitted by this {@link SmartContract} and returns an array of events with their corresponding types.
+    @async
+    @param [start=UInt32.from(0)] - The start height of the events to fetch.
+    @param end - The end height of the events to fetch. If not provided, fetches events up to the latest height.
+    @returns A promise that resolves to an array of objects, each containing the event type and event data for the specified range.
+    @throws If there is an error fetching events from the Mina network.
+    @example
+    const startHeight = UInt32.from(1000);
+    const endHeight = UInt32.from(2000);
+    const events = await myZkapp.fetchEvents(startHeight, endHeight);
+    console.log(events);
+  */
   async fetchEvents(
     start: UInt32 = UInt32.from(0),
     end?: UInt32
   ): Promise<{ type: string; event: ProvablePure<any> }[]> {
     // filters all elements so that they are within the given range
     // only returns { type: "", event: [] } in a flat format
-    let events = (await Mina.fetchEvents(this.address, this.self.body.tokenId))
+    let events = (
+      await Mina.fetchEvents(this.address, this.self.body.tokenId, {
+        from: start,
+        to: end,
+      })
+    )
       .filter((el: any) => {
```

Comment:
all those `any` coercions make me uncomfortable, if you have time and could remove them it would be great -- usually functions like `map` and `filter` infer the element type anyway, so you don't need a type signature

Reply:
Sounds good! Fixed typing :D